pkg/protoconv/resources/resources.go

```diff
@@ -84,6 +85,7 @@ func NewDeploymentFromStaticResource(obj interface{}, deploymentType, clusterID,
 
 	wrap := newWrap(objMeta, kind, clusterID, registryOverride)
 	wrap.populateFields(obj)
+	wrap.Deployment.Id = uuid.NewV4().String()
```

Comment:
Let's add a comment here explaining why we are adding random IDs in case someone encounters this in the future. 

Perhaps something like:
```
Deployment ID is empty because the processed yaml comes from roxctl and therefore doesn't 
get a  Kubernetes generated ID. This is a temporary ID only required for roxctl to distinguish 
between different generated deployments. 
```

Reply:
Okay, will do